src/aws/flb_aws_credentials.c

```diff
@@ -323,6 +324,183 @@ struct flb_aws_provider *flb_standard_chain_provider_create(struct flb_config
+    config_key_sts_endpoint = flb_sds_create_len(config_key_prefix, key_max_len);
+    strcpy(config_key_sts_endpoint + key_prefix_len, "sts_endpoint");
+    config_key_role_arn = flb_sds_create_len(config_key_prefix, key_max_len);
+    strcpy(config_key_role_arn + key_prefix_len, "role_arn");
+    config_key_external_id = flb_sds_create_len(config_key_prefix, key_max_len);
+    strcpy(config_key_external_id + key_prefix_len, "external_id");
+
+    /* AWS provider needs a separate TLS instance */
+    cred_tls = flb_tls_create(FLB_TRUE,
+                                    ins->tls_debug,
+                                    ins->tls_vhost,
+                                    ins->tls_ca_path,
+                                    ins->tls_ca_file,
+                                    ins->tls_crt_file,
+                                    ins->tls_key_file,
+                                    ins->tls_key_passwd);
+    if (!cred_tls) {
+        flb_errno();
+        goto error;
+    }
+
+    region = flb_output_get_property(config_key_region, ins);
+    if (!region) {
+        flb_plg_error(ins, "aws_auth enabled but %s not set", config_key_region);
+        goto error;
+    }
+
+    /* Use null sts_endpoint if none provided */
+    sts_endpoint = flb_output_get_property(config_key_sts_endpoint, ins);
+
+    aws_provider = flb_standard_chain_provider_create(config,
+                                                      cred_tls,
+                                                      (char *) region,
+                                                      (char *) sts_endpoint,
+                                                      NULL,
+                                                      flb_aws_client_generator());
+    if (!aws_provider) {
+        flb_plg_error(ins, "Failed to create AWS Credential Provider");
+        goto error;
+    }
+
+    role_arn = flb_output_get_property(config_key_role_arn, ins);
+    if (role_arn) {
+        /* Use the STS Provider */
+        base_aws_provider = aws_provider;
+        external_id = flb_output_get_property(config_key_external_id, ins);
+
+        session_name = flb_sts_session_name();
+        if (!session_name) {
+            flb_plg_error(ins, "Failed to create aws iam role "
+                        "session name");
+            goto error;
+        }
+
+        /* STS provider needs yet another separate TLS instance */
+        sts_tls = flb_tls_create(FLB_TRUE,
+                                            ins->tls_debug,
+                                            ins->tls_vhost,
+                                            ins->tls_ca_path,
+                                            ins->tls_ca_file,
+                                            ins->tls_crt_file,
+                                            ins->tls_key_file,
+                                            ins->tls_key_passwd);
+        if (!sts_tls) {
+            flb_errno();
```

Comment:
Could we add an error log for this error too?

Reply:
yep